<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <title>SequenceServer</title>
    <meta name="author" content="Anurag Priyam">
    <meta name="author" content="Yannick Wurm">
    <meta name="keywords" content="local NCBI BLAST, custom BLAST, local BLAST server, BLAST web frontend, local BLAST, BLAST web interface, galaxy, viroblast">
    <meta name="description" content="Use SequenceServer to pragmatically perform and interpret genomic comparisons using NCBI BLAST. SequenceServer accelerates your analyses so you can focus on the science.">
    <!-- Support HTML5 elements and media queries on IE8 and less. -->
    <!--[if lt IE 9]>
      <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.2/html5shiv.min.js"></script>
    <![endif]-->

    <!-- Bootstrap and FontAwesome CSS -->
    <link rel="stylesheet" type="text/css" href="/css/googlefont-source-sans-pro.css">
    <link rel="stylesheet" type="text/css" href="/css/bootstrap-3.3.6.min.css">
    <link rel="stylesheet" type="text/css" href="/css/fontawesome-4.3.0.css">
    <link rel="stylesheet" type="text/css" href="/css/pygments-github.css">
    <link rel="stylesheet" type="text/css" href="/css/seqserv.css">

    <!-- Google Analytics -->
    <script>
      (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
      })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

      ga('create', 'UA-471841-7', 'sequenceserver.com');
      ga('send', 'pageview');
    </script>
  </head>

  <body>
    <nav
      class="navbar navbar-fixed-top"
      role="navigation">
      <div
        class="container">
        <div
          class="navbar-header">
          <button
            type="button" class="navbar-toggle"
            data-toggle="collapse" data-target=".bs-navbar-collapse">
            <span
              class="sr-only">
              Toggle navigation
            </span>
            <i class="fa fa-bars"></i>
          </button>
          <div class="brand-container">
            <a
              class="navbar-brand navbar-left" href="/">
              <img class="logo" src="/img/logos/SequenceServer_logo.png" alt="sequenceserver_logo">
              &nbsp;<span>1.0.14</span>
            </a>
            <a
              class="navbar-brand-made-by"
              href="//wurmlab.github.io">
              created by Wurmlab
            </a>
          </div>
        </div>
        <div
          class="collapse navbar-collapse bs-navbar-collapse">
          <ul class="nav navbar-nav">
            <li>
              <a href="/#installation">
                Installation
              </a>
            </li>
            <li>
              <a href="/doc">
                Advanced configuration
              </a>
            </li>
            <li>
              <a href="/cloud" style="font-weight: bold;">
                Cloud service
              </a>
            </li>
            <li>
              <a href="/#license">
                License
              </a>
            </li>
            <li>
              <a href="/#support">
                Support
              </a>
            </li>
            <li>
              <a href="/#users">
                Users
              </a>
            </li>
            <li>
              <a class="visible-lg"
                href="https://antgenomes.sequenceserver.com"
                title="Will open in new window" target="_blank">
                <i class="fa fa-external-link" aria-hidden="true"></i>
                Demo
              </a>
            </li>
            <li>
              <a class="visible-lg"
                href="https://github.com/wurmlab/sequenceserver"
                title="Will open GitHub in new window" target="_blank">
                <i class="fa fa-external-link" aria-hidden="true"></i>
                Code
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>

    <div
      class="container">
      <h1 id="sequence-search">Sequence Search</h1>

<h2 id="why-search-sequences">Why search sequences?</h2>

<p>Reasons for performing sequence searches include:</p>

<ul>
  <li>wanting to know if the Sanger sequence of the gene you cloned indeed matches the sequence you were expecting,</li>
  <li>checking whether targeted CRISPR mutagenesis was successful,</li>
  <li>identifying the sequence of a candidate gene in a newly sequenced genome,</li>
  <li>finding out the potential function of a gene,</li>
  <li>finding paralogs or orthologs of a gene to perform multiple sequence alignment and identify which parts of the sequence are most conserved,</li>
  <li>searching whether oligonucleotide primer sequences are likely to amplify non-target regions of the genome.</li>
</ul>

<h2 id="where-to-perform-sequence-searches">Where to perform sequence searches</h2>

<p>The most widely used online portal for sequence searches is <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">NCBI’s BLAST search</a>. It’s a handy go-to place given that NCBI house the vast majority of published nucleotide and protein sequences.</p>

<p>However, other contexts for BLAST sequence search also exist. These include:
The <a href="https://www.ebi.ac.uk/ena/browser/sequence-search">European Nucleotide Archive</a>, which hosts a mirror of NCBI’s data,
<a href="https://www.uniprot.org/blast/">UniProt</a>, who have subsets of the protein sequence data that meet specific quality standards,
and domain-specific websites such as <a href="http://flybase.org">Flybase</a> or the <a href="https://antgenomes.org">ant genome database</a>.</p>

<p>Challenges with such major repositories include that they get a lot of demand. Thus your sequence search can be “queued” for a while until computing capacity becomes available. Furthermore, public websites for sequence search typically have size restrictions on query sequences. Finally, they do not enable you to search against unpublished sequences.</p>

<p>Most sequence search types can also be performed on a local computing cluster, as is found in many universities, research centers and core facilities and institutes.</p>

<p>Our SequenceServer software provides a pragmatic alternative for <a href="https://www.sequenceserver.com">performing local BLAST sequence searches</a> on your computer, including on unpublished data. You can install it and run it locally on a Mac or Linux (its free). But you’ll be limited to your local computing power. Alternatively,  you can use <a href="https://www.sequenceserver.com/cloud">SequenceServer Cloud</a>. Having a SequenceServer Cloud instance enables you (or your team) to have a centrally accessed sequence search repository. Its graphical sequence search interface is fast, accessible from any web browser (including from Windows), takes no space on your computer, and includes many features to facilitate your analysis.</p>

<div class="container">
  <div class="row justify-content-center">
 	<div class="alert alert-info">
	  <p>By leveraging cloud computing and publication-ready graphics, SequenceServer Cloud makes it easy to perform sequence search results and to interpret them. <a href="http://sequenceserver.com/cloud">Learn more</a></p>
	  <p style="text-align:center"><a href="https://sequenceserver.com/cloud"><img src="/img/logos/SequenceServer_logo.png" alt="Sequence Search with SequenceServer" width="200pt" /></a></p>
    </div>
  </div>
</div>

<h2 id="the-broad-diversity-of-sequence-search-algorithms">The broad diversity of sequence search algorithms</h2>

<p>BLAST, whether used at NCBI, as local installation, or <a href="https://www.sequenceserver.com/cloud">online using a cloud service</a> is the mainly used sequence search algorithm, with <strong>more than 100,000 citations</strong>.</p>

<p>BLAST is great for searching large databases with “small” query sequences. Today’s BLAST algorithm is far more computationally efficient than those from twenty years ago. However, BLAST isn’t necessarily the most appropriate sequence search algorithm for every job. Other algorithms include BLAT, USearch, minimap. The following article reviews the history of sequence search algorithms and the tradeoffs among search algorithms:</p>

<blockquote>
  <p><a href="https://doi.org/10.1093/bioinformatics/btz272"><strong>Evolution of biosequence search algorithms: a brief survey</strong></a><br />
<em>Gregory Kucherov</em>. Bioinformatics (2019), 35:3547–3552<br />
<em>Although modern high-throughput biomolecular technologies produce various types of data, biosequence data remain at the core of bioinformatic analyses. However, computational techniques for dealing with this data evolved dramatically.
Results
In this bird’s-eye review, we overview the evolution of main algorithmic techniques for comparing and searching biological sequences. We highlight key algorithmic ideas emerged in response to several interconnected factors: shifts of biological analytical paradigm, advent of new sequencing technologies and a substantial increase in size of the available data. We discuss the expansion of alignment-free techniques coming to replace alignment-based algorithms in large-scale analyses. We further emphasize recently emerged and growing applications of sketching methods which support comparison of massive datasets, such as metagenomics samples. Finally, we focus on the transition to population genomics and outline associated algorithmic challenges.</em></p>
</blockquote>

    </div>

    <footer
      style="background-color: white">
      <hr>
      <div
        class="container"
        style="text-align: center; padding: 20px 0;">
        <p>
          <a
            href="http://www.biorxiv.org/content/early/2015/11/27/033142">
            Priyam A, Woodcroft BJ, Rai V, Munagala A, Moghul I, Ter F, Gibbins MA, Moon H, Leonard G, Rumpf W &amp; Wurm Y. 2015. Sequenceserver: a modern graphical user interface for custom BLAST databases. biorxiv doi: 10.1101/033142.
          </a>
        </p>
        <p>
          SequenceServer is &copy; 5bases pvt ltd (2012-2016) and authors.
          Parts of SequenceServer are &copy; Universit&eacute; de Lausanne
          (2011).
        </p>
        <p>
          Archived site: <a
            href="http://www.sequenceserver.com/old/">
            SequenceServer 0.8 (pre March 2015)</a>.
        </p>
      </div>
    </footer>

    <!-- jQuery and Bootstrap JS -->
    <script src="/js/jquery-2.1.3.min.js"></script>
    <script src="/js/bootstrap-3.3.2.min.js"></script>
  </body>
</html>
